### ISSUE TITLE
Invalid opcode 199/8/8

### ISSUE THREAD

**Issue by uasan** [fiberphp/fiber-ext#5]
OS Fedora 26

php -v
```
PHP 7.3.0-dev (cli) (built: Jan  9 2018 16:04:55) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.0-dev, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.0-dev, Copyright (c) 1999-2018, by Zend Technologies
```

Run script in `php-cli`
```PHP
<?php
function sub1()
{
    // yield from sub call
    return Fiber::yield(1);
}
$fiber = new Fiber(function ($a, $b) {
    $c = Fiber::yield($a + $b);

    $d = sub1();
    return $d.$c;
});

echo $fiber->resume(1, 2);     // echo 3
echo $fiber->resume("world");  // echo 1
echo $fiber->resume("hello "); // echo "hello world"
```
Output
```
31
Fatal error: Invalid opcode 199/8/8. in  on line 0
```

Am I doing something wrong?

**Comment by lvht** [comment]
It seems that the latest master code of php-src breaks the ZEND_USER_OPCODE.

**Comment by lvht** [comment]
With the latest master branch of php-src, the `ZEND_VM_SET_OPCODE_HANDLER` macro seems not work for user opcode .
https://github.com/fiberphp/fiber-ext/blob/76c1a133a739f2afb0269ae5f6b5637f072d1b2a/fiber.c#L573-L579

If I replace the `ZEND_VM_SET_OPCODE_HANDLER` with
```c
fiber_terminate_op[0].handler = fiber_terminate_opcode_handler;
```
it works again. You @uasan make your own try.

However, I am not sure if this change has any other problem. 
Could you @dstogov offer any help? Thank you.

**Comment by uasan** [comment]
Today clone current version master `php-src`
```
PHP 7.3.0-dev (cli) (built: Jan 10 2018 23:12:01) ( NTS )
```
New problem, after call `$fiber->resume()`:
```
Fatal error: Allowed memory size of 134217728 bytes exhausted...
```

I think, try PHP 7.2 with a patch, maybe it works stably.

**Comment by uasan** [comment]
Yes, work stably in `PHP 7.2.1` and `end_fiber.patch`
```
PHP 7.2.1-dev (cli) (built: Jan 11 2018 04:41:34) ( NTS )
```

**Comment by lvht** [comment]
try fix by #11
